File: leetcode/easy/intersection-of-multiple-arrays.ts
// https://leetcode.com/problems/intersection-of-multiple-arrays/

// MINE: map
/*
function intersection(nums: number[][]): number[] {
    const map = new Map(), ans = []
    for(let A of nums) {
        for(let n of A) map.set(n, map.get(n)+1 || 1)
    }
    for(let [k, v] of map) if(v === nums.length) ans.push(k)
    return ans.sort((a, z) => a - z)
};
*/

// ANSWER: array > no sort
function intersection(nums: number[][]): number[] {
  const count = new Array(1001).fill(0),
    ans = []
  for (let A of nums) for (let n of A) ++count[n]
  for (let i = 0; i < count.length; ++i) if (count[i] === nums.length) ans.push(i)
  return ans
}

File: leetcode/easy/intersection-of-multiple-arrays.py
# https://leetcode.com/problems/intersection-of-multiple-arrays/

class Solution:
    def intersection(self, A: List[List[int]]) -> List[int]:
        # cnt, ans = [0]*1001, []
        # for A in nums:
        #     for n in A: cnt[n] += 1
        # for i, v in enumerate(cnt):
        #     if v == len(nums): ans.append(i)
        # return ans

        return sorted([k for k, v in Counter([x for l in A for x in l]).items() if v == len(A)])
